firebase-firestore/src/main/java/com/google/firebase/firestore/core/FirestoreClient.java

```diff
@@ -143,6 +145,10 @@ public FirestoreClient(
           if (gcScheduler != null) {
             gcScheduler.stop();
           }
+
+          if (indexScheduler != null && Persistence.INDEXING_SUPPORT_ENABLED) {
```

Comment:
Do we need to check the flag here? The scheduler would not be available if the flag is off.

The name is also a bit generic. I wonder if we should include "backfill" in the scheduler name, which reduces the mental scope from "all of indexing" to "backfill index data".